cookbooks/aws-parallelcluster-config/templates/default/directory_service/update_directory_service_password.sh.erb

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -e
+
+SSSD_CONFIG_FILE="/etc/sssd/sssd.conf"
+SSSD_SHARED_CONFIG_FILE="<%= @shared_sssd_conf_path %>"
+SSSD_SECTION="domain/default"
+SSSD_PROPERTY="ldap_default_authtok"
+SECRET_ARN="<%= @secret_arn %>"
+REGION="<%= @region %>"
+
+echo "Reading password from ${SSSD_CONFIG_FILE}"
+password_from_sssd_config=$(python3 -c "import configparser;file='${SSSD_CONFIG_FILE}';config=configparser.ConfigParser();config.read(file);print(config['${SSSD_SECTION}']['${SSSD_PROPERTY}'])")
+echo "Password from ${SSSD_CONFIG_FILE}: ${password_from_sssd_config}"
+
+echo "Reading password from AWS Secrets Manager: ${SECRET_ARN}"
+password_from_secrets_manager=$(aws secretsmanager get-secret-value --secret-id ${SECRET_ARN} --region ${REGION} --query 'SecretString' --output text)
+echo "Password from AWS Secrets Manager: ${password_from_secrets_manager}"
+
+[ "${password_from_sssd_config}" == "${password_from_secrets_manager}" ] && echo "Password match, skipping update" && exit 0
+
+echo "Writing AWS Secrets Manager password to ${SSSD_CONFIG_FILE}"
+python3 -c "import configparser;file='${SSSD_CONFIG_FILE}';config=configparser.ConfigParser();config.read(file);config['${SSSD_SECTION}']['${SSSD_PROPERTY}']='${password_from_secrets_manager}'; config.write(open(file,'w'))"
+password_from_sssd_config=$(python3 -c "import configparser;file='${SSSD_CONFIG_FILE}';config=configparser.ConfigParser();config.read(file);print(config['${SSSD_SECTION}']['${SSSD_PROPERTY}'])")
```

Comment:
this command is duplicated you can write a "_get_ssd_password" to use in both the places.

Reply:
Since I'm going to remove password printing, I'll also remove the second read.